cpp/include/proxy_sequences.h

```diff
@@ -129,6 +131,53 @@ class UpdateDeleteObject final : public Update {
 
   std::string _name;
 };
+
+BOOLEAN_ENUM(ObjectExists);
+
+class Sequence {
+public:
+  Sequence(size_t assert_size_hint = 0, size_t update_size_hint = 0)
+      : _asserts(assert_size_hint), _updates(update_size_hint) {}
+
```

Comment:
The size hints need to be used with the respective vector's `reserve` method - passing them to the constructor will lead to `size_hint` `nullptr`s. being added as the first elements, and `add_assert` and friends then push new entries behind these `nullptr`s.